_posts/2020-04-13-Julia-Antipatterns.md

```diff
@@ -0,0 +1,831 @@
+---
+layout: default
+title: "Julia-Antipatterns"
+tags:
+    - julia
+    - jupyter-notebook
+---
+An antipattern is a common solution to a problem that over-all makes things worse than they could have been.
+This blog post aims to highlight a few antipatterns common in Julia code.
+I suspect a lot of this is due to baggage brought from other languages, where these are not Antipatterns, but are in fact good patterns.
+This post is to clear things up.
+<!--more-->
+
+This post will not cover anything in the [Julia Performance Tips](https://docs.julialang.org/en/v1/manual/performance-tips/).
+
+Each section in this will describe a different antipattern.
+It will discuss:
+ - Examples, including highlighting this issues
+ - What to do instead
+ - Reasons people do this
+
+## NotImplemented Exceptions
+
+This shows up in packages defining APIs that others should implement.
+Where one wants to declare and document functions that implementors of the API should overload.
+In this antipattern, one declares an abstract type, and a function that takes that abstract type as an argument.
+Usually, as the the first abstract following something like fashion from Python etc, as taking the object as the first argument.
```

Comment:
Should this be:
Usually, as the first argument following something like the fashion from Python etc, as taking the object as the first argument.